hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -651,6 +651,15 @@ object DataSourceWriteOptions {
     .sinceVersion("0.14.0")
     .withDocumentation("Controls whether spark sql prepped update, delete, and merge are enabled.")
 
+  val OVERWRITE_MODE: ConfigProperty[String] = ConfigProperty
+    .key("hoodie.datasource.overwrite.mode")
```

Comment:
Why not define this config to `overwrite.dynamic.partition`? We just distingish the dynamic partition and the whole table for insert overwrite. The static mode is clear and no need to config.